tests/datastore_mysql/test_database.py

```diff
@@ -0,0 +1,165 @@
+        ('Datastore/statement/MySQL/datastore_mysql_%s/delete' % DB_NAMESPACE, 1),
+        ('Datastore/operation/MySQL/drop', 2),
+        ('Datastore/operation/MySQL/create', 2),
+        ('Datastore/statement/MySQL/hello/call', 1),
+        ('Datastore/operation/MySQL/commit', 2),
+        ('Datastore/operation/MySQL/rollback', 1)]
+
+_test_execute_via_cursor_rollup_metrics = [
+        ('Datastore/all', 13),
+        ('Datastore/allOther', 13),
+        ('Datastore/MySQL/all', 13),
+        ('Datastore/MySQL/allOther', 13),
+        ('Datastore/operation/MySQL/select', 1),
+        ('Datastore/statement/MySQL/datastore_mysql_%s/select' % DB_NAMESPACE, 1),
+        ('Datastore/operation/MySQL/insert', 1),
+        ('Datastore/statement/MySQL/datastore_mysql_%s/insert' % DB_NAMESPACE, 1),
+        ('Datastore/operation/MySQL/update', 1),
+        ('Datastore/statement/MySQL/datastore_mysql_%s/update' % DB_NAMESPACE, 1),
+        ('Datastore/operation/MySQL/delete', 1),
+        ('Datastore/statement/MySQL/datastore_mysql_%s/delete' % DB_NAMESPACE, 1),
+        ('Datastore/statement/MySQL/hello/call', 1),
+        ('Datastore/operation/MySQL/call', 1),
+        ('Datastore/operation/MySQL/drop', 2),
+        ('Datastore/operation/MySQL/create', 2),
+        ('Datastore/operation/MySQL/commit', 2),
+        ('Datastore/operation/MySQL/rollback', 1)]
+
+@validate_transaction_metrics('test_database:test_execute_via_cursor',
+        scoped_metrics=_test_execute_via_cursor_scoped_metrics,
+        rollup_metrics=_test_execute_via_cursor_rollup_metrics,
+        background_task=True)
+@validate_database_trace_inputs(sql_parameters_type=dict)
+@background_task()
+def test_execute_via_cursor(table_name):
+
+    connection = mysql.connector.connect(db=DB_SETTINGS['name'],
+            user=DB_SETTINGS['user'], passwd=DB_SETTINGS['password'],
+            host=DB_SETTINGS['host'], port=DB_SETTINGS['port'])
+
+    cursor = connection.cursor()
+
+    cursor.execute("""drop table if exists `%s`""" % table_name)
+
+    cursor.execute("""create table %s """
+            """(a integer, b real, c text)""" % table_name)
+
+    cursor.executemany("""insert into `%s` """ % table_name +
+            """values (%(a)s, %(b)s, %(c)s)""", [dict(a=1, b=1.0, c='1.0'),
+            dict(a=2, b=2.2, c='2.2'), dict(a=3, b=3.3, c='3.3')])
+
+    cursor.execute("""select * from %s""" % table_name)
+
+    for row in cursor: pass
+
+    cursor.execute("""update `%s` """ % table_name +
+            """set a=%(a)s, b=%(b)s, c=%(c)s where a=%(old_a)s""",
+            dict(a=4, b=4.0, c='4.0', old_a=1))
+
+    cursor.execute("""delete from `%s` where a=2""" % table_name)
+
+    cursor.execute("""drop procedure if exists hello""")
+    cursor.execute("""CREATE PROCEDURE hello()
+                      BEGIN
+                        SELECT 'Hello World!';
+                      END""")
+
+    cursor.callproc("hello")
```

Comment:
You'll need to add the PID to this procedure name as well to get around concurrency issues.